lwgenerate/gcp/gke.go

```diff
@@ -0,0 +1,213 @@
+package gcp
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/lacework/go-sdk/lwgenerate"
+	"github.com/pkg/errors"
+)
+
+type GenerateGkeTfConfigurationArgs struct {
+	ExistingServiceAccount    *lwgenerate.ServiceAccount
+	ExistingSinkName          string
+	IntegrationName           string
+	Labels                    map[string]string
+	LaceworkProfile           string
+	OrganizationId            string
+	OrganizationIntegration   bool
+	Prefix                    string
+	ProjectId                 string
+	PubSubSubscriptionLabels  map[string]string
+	PubSubTopicLabels         map[string]string
+	ServiceAccountCredentials string
+	WaitTime                  string
+}
+
+type Modifier func(c *GenerateGkeTfConfigurationArgs)
+
+func (args *GenerateGkeTfConfigurationArgs) Generate() (string, error) {
+	if err := args.validate(); err != nil {
+		return "", errors.Wrap(err, "invalid inputs")
+	}
+
+	requiredProviders, err := createRequiredProviders()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate required providers")
+	}
+
+	gcpProvider, err := createGcpProvider(args.ServiceAccountCredentials, args.ProjectId)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate gcp provider")
+	}
+
+	laceworkProvider, err := createLaceworkProvider(args.LaceworkProfile)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate lacework provider")
+	}
+
+	gkeAuditLogModule, err := createGKEAuditLog(args)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate GKE Audit Log module")
+	}
+
+	hclBlocks := lwgenerate.CreateHclStringOutput(
+		lwgenerate.CombineHclBlocks(
+			requiredProviders,
+			gcpProvider,
+			laceworkProvider,
+			gkeAuditLogModule),
+	)
+	return hclBlocks, nil
+}
+
+func (args *GenerateGkeTfConfigurationArgs) validate() error {
+	if args.OrganizationIntegration && args.OrganizationId == "" {
+		return errors.New("An Organization ID must be provided for an Organization Integration")
+	}
+
+	if !args.OrganizationIntegration && args.OrganizationId != "" {
+		return errors.New("To provide an Organization ID, Organization Integration must be true")
```

Comment:
```suggestion
		return errors.New("to provide an Organization ID, Organization Integration must be true")
```

https://github.com/golang/go/wiki/CodeReviewComments#error-strings